presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/ResourceGroupSpec.java

```diff
@@ -164,6 +173,21 @@ public ResourceGroupNameTemplate getName()
         return hardCpuLimit;
     }
 
+    public Optional<Duration> getPerQueryExecutionTimeLimit()
+    {
+        return perQueryExecutionTimeLimit;
+    }
+
+    public Optional<DataSize> getPerQueryTotalMemoryLimit()
+    {
+        return perQueryTotalMemoryLimit;
+    }
+
+    public Optional<Duration> getPerQueryCpuTimeLimit()
+    {
+        return perQueryCpuTimeLimit;
+    }
```

Comment:
I think it would be great to put these into a separate inner structure, let's call it `perQueryLimits`, and rename the inner values things like `executionTimeLimit`, `totalMemoryLimit`, and `cpuTimeLimit`.  The reason is, everything else in the resource group is aggregated via `SUM`, whereas these are aggregated via `MIN`, and the visual separation I think would be nice to clearly separate these two things (more than the naming in the field itself).

Reply:
Another area where we went back and forth on the implementation :) We finally settled on keeping the flat structure since it goes better with the overall design of the current class. The per query limits are not aggregated at the RG level, it is just a metadata that is sent back to the coordinator. I consider it more close to `schedulingWeight` - which is just a metadata. This also makes it closer to the 1 column per field approach for the DB based implementation for instance and in general just makes it easier to declare these limits in the configuration.

For me having this as a separate field might have made sense if the structure were similar to:

```
ResourceGroup {
    name: "foo",
    schedulingPolicy: "",
    schedulingWeight: "",
    concurrencyControls: {
        maxConcurrency: "",
        maxQueued: "",
        hardCpuLimit: "",
    },
    perQueryLimits: {
        ....
    }
}
```

That way there was a clear differentiation between metadata vs concurrency controls vs per query limits. Just having per query limits as a different field seemed against the current pattern to me.